pyqt-apps/siriushla/as_rf_control/control.py

```diff
@@ -883,42 +894,48 @@ def _rampMonLayout(self):
         self.curve_vgav.setVisible(False)
         self.curve_pwr.setVisible(False)
 
-        self.lb_vgapdesc = QLabel(
-            '<h4>Gap Voltage:</h4>', self, alignment=Qt.AlignCenter)
-
-        self.lb_c3pwrbot = SiriusLabel(
-            self, self.prefix+'BO-05D:RF-P5Cav:Cell3PwrBot-Mon')
-        self.lb_c3pwrbot.showUnits = True
-        self.lb_pwrfwdbot = SiriusLabel(
-            self, self.prefix+'BO-05D:RF-P5Cav:PwrFwdBot-Mon')
-        self.lb_pwrfwdbot.showUnits = True
-        self.lb_pwrrevbot = SiriusLabel(
-            self, self.prefix+'BO-05D:RF-P5Cav:PwrRevBot-Mon')
-        self.lb_pwrrevbot.showUnits = True
+        self.cb_ramp = QComboBox()
+        self.cb_ramp.setStyleSheet("QComboBox{max-width: 3.5em; font-weight: bold;}")
+        self.cb_ramp.addItems(["W", "mV"])
+        self.cb_ramp.currentTextChanged.connect(self._handle_rmptab_visibility)
+
         self.lb_c3phsbot = SiriusLabel(
             self, self.prefix+'BR-RF-DLLRF-01:BOT:CELL3:PHS')
         self.lb_c3phsbot.showUnits = True
         self.lb_cavvgapbot = SiriusLabel(
             self, self.prefix+'BO-05D:RF-P5Cav:RmpAmpVCavBot-Mon')
         self.lb_cavvgapbot.showUnits = True
 
-        self.lb_c3pwrtop = SiriusLabel(
-            self, self.prefix+'BO-05D:RF-P5Cav:Cell3PwrTop-Mon')
-        self.lb_c3pwrtop.showUnits = True
-        self.lb_pwrfwdtop = SiriusLabel(
-            self, self.prefix+'BO-05D:RF-P5Cav:PwrFwdTop-Mon')
-        self.lb_pwrfwdtop.showUnits = True
-        self.lb_pwrrevtop = SiriusLabel(
-            self, self.prefix+'BO-05D:RF-P5Cav:PwrRevTop-Mon')
-        self.lb_pwrrevtop.showUnits = True
+
+        lay = QGridLayout()
+        self.ramp_chn_wid = {}
+
+        for unit_type, rmp_channels in self.chs['Ramp'].items():
+            idy = 1
+            self.ramp_chn_wid[unit_type] = {}
+            for pos, rmp_ch_pvs in rmp_channels.items():
```

Comment:
consider making:
```python3
for idy, (pos, ramp_ch_pvs) in enumerate(rmp_channels.items(), 1):
```
instead of defining and incrementing these extra variables.